crates/criticalup-cli/tests/cli/clean.rs

```diff
@@ -255,3 +266,93 @@ async fn removes_unused_installations_from_disk_that_do_not_have_state() {
     assert!(!toolchains_dir.join(installation_id_2).exists()); // Does not exist.
     assert!(!toolchains_dir.join(installation_id_3).exists()); // Does not exist.
 }
+
+/// Remove the installation from the state, if the installation directory does not exist. This is
+/// true even if the installation in the state has a manifest.
+#[tokio::test]
+async fn clean_deletes_only_unused_installations_that_are_not_on_disk() {
+    let test_env = TestEnvironment::prepare().await;
+    let root = test_env.root();
+    let toolchains_dir = root.join("toolchains");
+    fs::create_dir_all(&toolchains_dir).unwrap();
+
+    let installation_id_1 = "installation_id_1";
+    let installation_id_2 = "installation_id_2";
+    let installation_id_3 = "installation_id_3";
+
+    let state_file_in_root = root.join("state.json");
+    let mut state_file = std::fs::File::create(&state_file_in_root).unwrap();
+    let content = json!({
```

Comment:
I don't love that this is effectively [`StateRepr`](https://github.com/ferrocene/criticalup/blob/fee32bcc776d9a4bcccd9abd2ff1b7d8b66eb2ba/crates/criticalup-core/src/state.rs#L345) but untyped. :(

But I understand why.